### DISCUSSION TITLE
Apple M1 silicone coherence problem

### DISCUSSION THREAD

**Discussion by Daniel3356** [abrahamjuliot/creepjs#163] [Q&A] [answered]
I have an Apple M1, Creepjs detect my mac as ARM with is true, but indicate as an Intel CPU. 
![browser](https://user-images.githubusercontent.com/15313941/147303724-5d4f8def-6e52-4930-a2c9-40ea236fa6a5.jpg)

**Comment by abrahamjuliot** [comment, marked as answer]
The userAgentData section gets the real `platform`, `version`, and `architecture` (macOS 12.0.1 arm), and the device section gets metrics that no longer provide updated platform info.

For example, log this to the console
```js
let {
  platform: uaPlatform,
  architecture,
  platformVersion
} = await navigator.userAgentData.getHighEntropyValues(['architecture', 'platform', 'platformVersion'])
let { userAgent, platform } = navigator

console.log('userAgentData:', uaPlatform, platformVersion, architecture)
console.log('navigator: ', /\(([^\)]+)/g.exec(userAgent)[1], platform)
```

The device section gets metrics from `navigator.userAgent` and `navigator.platform`. Starting in Chrome 101, those metrics will be reduced across all platforms. 

https://www.chromium.org/updates/ua-reduction

`webgl` should get the "Apple M1" chip on macOS Monterey.
```js
gl = new OffscreenCanvas(0,0).getContext('webgl')
ext = gl.getExtension('WEBGL_debug_renderer_info')
renderer = gl.getParameter(ext.UNMASKED_RENDERER_WEBGL)
console.log(renderer)
```

**Comment by abrahamjuliot** [comment]
closing for now, but feel free to discuss
